openstack/baremetal/v1/nodes/doc.go

```diff
@@ -0,0 +1,65 @@
+package nodes
+
+/*
+Package nodes provides information and interaction with the nodes API
+resource in the OpenStack Bare Metal service.
+
+	// Example to List Nodes
+	nodes.List(client, nodes.ListOpts{
+		Detail: true,
+		ProvisionState: Deploying,
+	}).EachPage(func(page pagination.Page) (bool, error) {
+		nodeList, err := nodes.ExtractNodes(page)
+		if err != nil {
+			return false, err
+		}
+
+		for _, n := range nodeList {
+			// Do something
+		}
+
+		return true, nil
+	})
+
+	// Example to Create Node
+	createNode, err := nodes.Create(client, nodes.CreateOpts{
+		Driver:        "ipmi",
+		BootInterface: PXEBoot,
```

Comment:
There was a discussion about this on an earlier version of the PR - https://github.com/gophercloud/gophercloud/pull/1431#discussion_r253270627, which I decided against because I think new ones could be easily introduced by drivers, right?

In the end, I basically came down to driver supplied values to be as flexible as possible, but Ironic core like provisioning states could be constants - does that make sense?

Reply:
(I'll update the docs unless you think PXEBoot/DiskBoot/etc would be good to have)

Reply:
I think we should allow arbitrary things, so let's fix the docs.